app/src/main/java/org/fossasia/susi/ai/chat/ParseSusiResponseHelper.kt

```diff
@@ -65,19 +68,43 @@ class ParseSusiResponseHelper {
             }
 
             Constant.VIDEOPLAY -> try {
-               // answer = susiResponse.answers[0].actions[i].expression
+                // answer = susiResponse.answers[0].actions[i].expression
                 identifier = susiResponse.answers[0].actions[i].identifier
             } catch (e: Exception) {
                 Timber.e(e)
             }
 
             Constant.AUDIOPLAY -> try {
-              //  answer = susiResponse.answers[0].actions[i].expression
+                //  answer = susiResponse.answers[0].actions[i].expression
                 identifier = susiResponse.answers[0].actions[i].identifier
             } catch (e: Exception) {
                 Timber.e(e)
             }
 
+            Constant.TABLE -> try {
+                val listColumn = ArrayList<String>()
+                val listColVal = ArrayList<String>()
+                val listTableData = ArrayList<String>()
+
+                susiResponse.answers.forEach {
+                    it.actions.forEach {
+                        it.columns?.forEach {
+                            listColumn.add(it.key)
+                            listColVal.add(it.value.toString())
+                        }
+                    }
+                    it.data.forEach {
+                        listColumn.forEach { i ->
+                            String
+                            listTableData.add(it[i].toString())
```

Comment:
This logic is strange to me, what is this doing?

Reply:
Consider an example of table type response, "Barcelona team players" this logic takes the column names first and then parses the blocks of data with the column keys and stores there value, here is the link for the response :https://api.susi.ai/susi/chat.json?timezoneOffset=-330&q=barcelona+team+players

Reply:
OK, I am approving this but this needs serious refactoring and tests

Reply:
I understand @iamareebjamal. I will look into this